internal/biz/usecase/resources/resource_service.go

```diff
@@ -822,3 +841,91 @@ func computeReadAfterWrite(uc *Usecase, write_visibility v1beta2.WriteVisibility
 	}
 	return !common.IsNil(uc.ListenManager) && uc.Config.ReadAfterWriteEnabled && isSPInAllowlist(reporterPrincipal, uc.Config.ReadAfterWriteAllowlist)
 }
+
+// getCurrentWorkspaceIdFromDatabase queries the database for the current workspace ID of a resource
+func (uc *Usecase) getCurrentWorkspaceIdFromDatabase(tx *gorm.DB, reporterResourceKey model.ReporterResourceKey) (string, error) {
```

Comment:
1. Any interaction with the DB should always be in the repository layer. The resource_service is an application service, so it should only have cross cutting concerns like interaction between domain models, transactions security etc. 
2. We already have a find repository method to lookup an existing resource, so I would be very careful about adding specific find methods unless absolutely necessary.
3. workspaceId to localResourceId is one of the many relations that we will need to replicate to spicedb, so this implementation is very specific to the current requirement, but our implementation needs to be such that we can calculate multiple tuples for a given resource.